src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectSystem/DefaultRazorProjectService.cs

```diff
@@ -177,11 +177,16 @@ public override void RemoveDocument(string filePath)
 
         if (!projectSnapshot.DocumentFilePaths.Contains(textDocumentPath, FilePathComparer.Instance))
         {
-            _logger.LogInformation("Containing project is not tracking document '{filePath}", filePath);
+            _logger.LogInformation("Containing project is not tracking document '{filePath}'", filePath);
+            return;
+        }
+
+        if (projectSnapshot.GetDocument(textDocumentPath) as DefaultDocumentSnapshot is not { } document)
```

Comment:
Nit, but this reads nicer to me
```suggestion
        if (projectSnapshot.GetDocument(textDocumentPath) is DefaultDocumentSnapshot document)
```

Reply:
:sound of forehead smack:

Oops. I was moving fast. You're 100% right and I'll fix those up. :smile: